src/ui/Controls/NikseComboBox.cs

```diff
@@ -236,20 +237,28 @@ public string SelectedText
             }
         }
 
-        protected override void OnGotFocus(EventArgs e)
+        private TaskScheduler _uiTaskScheduler;
+        protected override async void OnGotFocus(EventArgs e)
         {
             base.OnGotFocus(e);
             if (_textBox != null && DropDownStyle == ComboBoxStyle.DropDown)
             {
-                try
-                {
-                    Application.DoEvents();
-                    System.Threading.SynchronizationContext.Current.Post(TimeSpan.FromMilliseconds(25), () => _textBox.Focus());
```

Comment:
Btw, is there s specific reason why you want to delay 25ms before trigger the focus action?